### PR TITLE
Build lemon shared on windows

### LINKED ISSUE #15: lemon currently shipped as debug build on windows
see [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=184711&view=logs&j=f97ca392-d626-52d0-4b2e-4b27aa319a77&t=7b750183-140d-50ae-3ee1-cbef6218bdf6&l=1272) on azure pipelines

In `cmake --build . --config Release --target install` the `--config` option seems to be ignored.

### PR DESCRIPTION
**Pull request by k-dominik**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed) (version has not changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [] Ensured the license file is being packaged.

This PR enables shared libraries on windows by adding definitions for export macro. Also some minor fixes that were needed to get it running.

Tested as dependency of vigra with success (locally). 

Small note: I set the build number to 10 to make it more easily distinguishable from previous static builds.

See also #17 
Fixes #15

CC @hmaarrfk

### CODE REVIEW

**Comment by k-dominik** [comment]
@conda-forge-admin please re-render.

**Comment by k-dominik** [comment]
alternatively, one could of course delete all lemon windows packages up to now (they should not be functioning properly anyway).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,12 +29,13 @@ source:
    # library paths, even if BUILD_SHARED_LIBS=1
    # This patch fixes that.
     - 0003-LEMONConfig.cmake.patch
+   # export symbols for shared lib on windows
+    - 0004-allow-symbols-to-be-exported-in-shard-libs-on-win.patch  # [win]
 
 build:
-  number: 2
+  number: 10
```
Comment (hmaarrfk) [inline recipe/meta.yaml:36]:
```suggestion
  number: 3
```

**Comment by hmaarrfk** [comment]
@conda-forge/lemon any feedback at this stage?

File: recipe/meta.yaml (hunk 1 above)
Comment (k-dominik) [reply recipe/meta.yaml:36]: > Small note: I set the build number to 10 to make it more easily distinguishable from previous static builds.

so you're not a fan? :+1:

**Comment by k-dominik** [comment]
> @conda-forge/lemon any feedback at this stage?

in terms of feedback I can

**Comment by k-dominik** [comment]
So, what can I say. I've used lemon build with this PR when building vigra for our channel (ilastik-forge) and included it in our latest distribution of ilastik. So far, the parts we are interested in from lemon work. Should we just merge it?

@hmaarrfk what about the build number? I'm looking for an easy way to distinguish shared build vs the old one. But maybe build number is not the right tool for that

**Comment by hmaarrfk** [comment]
Build number is the correct way to distinguish

**Comment by k-dominik** [comment]
> Build number is the correct way to distinguish

aye, and you'd say just incrementing it by one is enough (I'm fine with this, just want to clarify).

Would you say we can merge this?

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:36]: right, yeah, just keep it simple. it will get reset to on then next update anyway

**Comment by hmaarrfk** [comment]
I think just incrementing is more than enough.

**Comment by hmaarrfk** [comment]
I can't merge this in.

@conda-forge/lemon this is ready for a final review.

**Comment by k-dominik** [comment]
I can merge it, would just like a second "yes" before doing it :+1: I'll change the build number to 3 then

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,12 +29,13 @@ source:
    # library paths, even if BUILD_SHARED_LIBS=1
    # This patch fixes that.
     - 0003-LEMONConfig.cmake.patch
+   # export symbols for shared lib on windows
+    - 0004-allow-symbols-to-be-exported-in-shard-libs-on-win.patch  # [win]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:33]: I guess there is no way to submit these patches upstream easily. we should get on mailing list and submit there maybe

**Comment by hmaarrfk** [comment]
It looks good to me, modulo the fact that I would like to see these accompanied with a PR or mailing list message attempting to upstream them.

File: recipe/meta.yaml (hunk 1 above)
Comment (k-dominik) [reply recipe/meta.yaml]: i don't think we can realistically expect a version bump in lemon :D

File: recipe/meta.yaml (hunk 2 above)
Comment (k-dominik) [reply recipe/meta.yaml:33]: oh wow, I was just checking and the website is back up - the issue tracker is active also, so there might be a way to sneak it in. In the meantime I'll merge this PR

**Pull request merged**